src/object_cmd.cpp

```diff
@@ -347,21 +350,100 @@ CommandCost CmdBuildObject(TileIndex tile, DoCommandFlag flags, uint32 p1, uint3
 			/* This may never be constructed using this method. */
 			return CMD_ERROR;
 
-		default: // i.e. NewGRF provided.
+		default: { // i.e. NewGRF provided.
+			const ObjectSpec *spec = ObjectSpec::Get(type);
```

Comment:
There is already a variable `spec` in this function, and it already has this value.
There are also `size_x` and `size_y` already.